pilot/pkg/autoregistration/controller.go

```diff
@@ -236,6 +239,25 @@ func (c *Controller) RegisterWorkload(proxy *model.Proxy, conTime time.Time) err
 	return err
 }
 
+// ensureProxyCanControlEntry ensures the connected proxy's identity matches that of the WorkloadEntry it is associating with.
+func ensureProxyCanControlEntry(proxy *model.Proxy, wle *config.Config) error {
+	if !features.ValidateWorkloadEntryIdentity {
+		// Validation disabled, skip
+		return nil
+	}
+	if proxy.VerifiedIdentity == nil {
+		return fmt.Errorf("registration of WorkloadEntry requires a verified identity")
+	}
+	if proxy.VerifiedIdentity.Namespace != wle.Namespace {
```

Comment:
have already checked this in DiscoveryServer.authorize

Reply:
I think in current code this is _implicitly_ checked but doesn't hurt to have defense in depth here since this is a CVE - we should explicitly check, even reduntly